src/main/java/com/google/alpollo/SongInfoServlet.java

```diff
@@ -0,0 +1,45 @@
+package com.google.alpollo;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.alpollo.model.SongInfo;
+import com.google.gson.Gson;
+
+/** Servlet created to deal with songInfo related operations. */
+@WebServlet("/song-info")
+public class SongInfoServlet extends HttpServlet {
+  private static final String SONG_ID = "id";
+  private static final String SONG_INFO = "songInfo";
+  private final Gson gson = new Gson();
+
+  /**
+   * Making a GET request to our servlet with the desired song ID as a parameter
+   * will return that song from the database.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String songId = request.getParameter(SONG_ID);
+    SongInfo songInfo = SongDataBase.getSongInfo(songId);
```

Comment:
This string won't compile, because on this branch getSongInfo expects `int` instead of `String`. You can fix on `songInfo-class` branch and upload this changes to this branch

Reply:
Why there you get the same id? I think in this case we need to use the same static method to get id from the song name and artist, but I didn't see it in the js code. It is more question to Olya. Why we have right id here?

Reply:
Olya can do a get request here to get a song by it's id, wasn't that the idea? Maybe I didn't understand.

Reply:
We discussed it today, so I think that it's solved (if not, let's discuss it in chat)